src/test/java/bio/terra/service/snapshotbuilder/SnapshotBuilderServiceTest.java

```diff
@@ -590,4 +593,71 @@ void notifySnapshotReady() {
         .snapshotReady(
             eq(id), anyString(), eq(snapshot.getName()), eq("No snapshot specification found"));
   }
+
```

Comment:
All of these tests have a common setup component that could be shared. TDR doesn't use it much, but there's a `@Nested` annotation that lets you group together tests that have a common state to avoid duplication. For example, this is one way to use it to rewrite the first test:
```
  @Nested
  class GroupMemberApis {
    SnapshotAccessRequestModel requestModel =
        SnapshotBuilderTestData.createAccessRequestModelApproved();
    UUID requestId = new SnapshotAccessRequestResponse().getId();

    @BeforeEach
    void beforeEach() {
      when(snapshotRequestDao.getById(requestId)).thenReturn(requestModel);
    }

    @Test
    void getGroupMembers() {
      SnapshotAccessRequestMembersResponse expected =
          new SnapshotAccessRequestMembersResponse().members(List.of());
      when(iamService.getGroupPolicyEmails(requestModel.samGroupName(), IamRole.MEMBER.toString()))
          .thenReturn(expected.getMembers());
      assertThat(snapshotBuilderService.getGroupMembers(requestId), is(expected));
    }
```

Reply:
The setup is different for the checking the case that the sam group name is null, so I put a Nested in that Nested, but I don't know if that's the appropriate way to use Nested/BeforeEach, so let me know if I should change it

Reply:
Interesting, I hadn't thought of multiple nestings like this. I think this is a good idea, one possible refinement is to to re-use the ID to avoid having a different ID in the nested cases. You can do this because `when()` will replace any existing `when()`s with the new mock behavior:
```

    @Nested
    class BlankSamGroup {
      @BeforeEach
      void beforeEach() {
        when(snapshotRequestDao.getById(requestId))
            .thenReturn(SnapshotBuilderTestData.createAccessRequestModelApproved());
      }

      @Test
      void testGetGroupMembersNotCreated() {
        assertThrows(
            IllegalArgumentException.class,
            () -> snapshotBuilderService.getGroupMembers(requestId));
      }
```

Or probably better, take the inner `BlankSamGroup` class and put it at the same scope as `GroupMemberApis`, since its tests don't use any of the state defined in `GroupMemberApis.